src/Workspaces/Core/Portable/FindSymbols/Declarations/DeclarationFinder.cs

```diff
@@ -76,23 +62,14 @@ internal static partial class DeclarationFinder
 
                 var symbolsWithName = symbols.ToImmutableArray();
 
-                if (startingCompilation != null && startingAssembly != null && !Equals(compilation.Assembly, startingAssembly))
-                {
-                    // Return symbols from skeleton assembly in this case so that symbols have 
-                    // the same language as startingCompilation.
-                    symbolsWithName = symbolsWithName.Select(s => s.GetSymbolKey(cancellationToken).Resolve(startingCompilation, cancellationToken: cancellationToken).Symbol)
-                                                     .WhereNotNull()
-                                                     .ToImmutableArray();
-                }
-
                 list.AddRange(FilterByCriteria(symbolsWithName, filter));
             }
         }
 
         private static async Task AddMetadataDeclarationsWithNormalQueryAsync(
             Project project,
-            IAssemblySymbol assembly,
-            PortableExecutableReference? reference,
+            AsyncLazy<IAssemblySymbol?> lazyAssembly,
```

Comment:
changed to lazyAssembly.  Avoids creation compilation when no results are found.